rspace/src/main/scala/coop/rchain/rspace/CloseOps.scala

```diff
@@ -0,0 +1,30 @@
+package coop.rchain.rspace
+
+trait CloseOps[T] {
+
+  @volatile private[this] var isClosed = false
```

Comment:
@ys-pyrofex have you tried running the perf-harness against it? I think it's a good view into the impact this might have on performance.

Reply:
isClosed variable updates only once at the end of RSpace life. Without competition between variable readers and writers volatile reads are free on x86 and x64 architecture ( more details here: https://brooker.co.za/blog/2012/09/10/volatile.htm ).

Since we're already closing the store, we may use non-volatile variable - I don't expect that result of store state consistency will be affected. The situation is identical to "transaction is already started on another thread when close() was called".
However Intel specs reserve right to do not propagate non-volatile variable updates to another processors for significant (up to infinite) amount of time on multi-processor systems (multi-socket motherboards). In real world, I suppose, this time will hardly be above a second, but who knows how situation will change for future systems.

To ensure I also launched some benchmarks. perf-harness shown a some deviation of results on my machine to understand if there is a slowdown or not. I launched rspace-bench to check pure if produce/consume performance is affected. My current tests show ~25000 events per second with 1.5% deviation for both volatile and non-volatile implementation.

As a result I recommend leave isClosed volatile.

Reply:
1) The link is dead (404 not found)
2) Which benchmarks did you exactly run? Can you write the commands here and paste the results that you got?

Reply:
1, Sorry, somehow lost 'l' at the end: https://brooker.co.za/blog/2012/09/10/volatile.html
2. The project rspace-bench has some example command lines to launch microbenchmarks, for example from IDEA you can run single-threaded test with: "rspaceBench/jmh:run EvalBench.reduceMVCEPPST -i 1 -wi 0 -f 0 -t 1"
I'm out of office right now, will show my results tomorrow morning

Reply:
@ys-pyrofex 
`-i -t 1` will not even allow jvm to warm properly
Can you paste those results you've mentioned?

Reply:
That was just an example, since Dominik and Lukasz have more experience with jmh than me I assumed that they know arguments that will better suit their PCs. I run tests with all resource-consuming apps, including browsers, closed.
My usual args for multi-threaded test are "EvalBench.reduceMVCEPPMT -i 10 -wi 5 -f 0 -t 4". Below are results of latest version made on another pc (not one mentioned in first review):

non-volatile:
[info] # Warmup Iteration   2: 119,236 ops/s
[info] # Warmup Iteration   3: 129,518 ops/s
[info] # Warmup Iteration   4: 139,749 ops/s
[info] # Warmup Iteration   5: 132,941 ops/s
[info] Iteration   1: 134,823 ops/s
[info] Iteration   2: 144,952 ops/s
[info] Iteration   3: 131,773 ops/s
[info] Iteration   4: 130,896 ops/s
[info] Iteration   5: 131,023 ops/s
[info] Iteration   6: 142,960 ops/s
[info] Iteration   7: 130,987 ops/s
[info] Iteration   8: 128,043 ops/s
[info] Iteration   9: 162,940 ops/s
[info] Iteration  10: 136,682 ops/s

volatile:
info] # Warmup Iteration   2: 135,020 ops/s
[info] # Warmup Iteration   3: 149,600 ops/s
[info] # Warmup Iteration   4: 140,408 ops/s
[info] # Warmup Iteration   5: 141,212 ops/s
[info] Iteration   1: 153,869 ops/s
[info] Iteration   2: 144,366 ops/s
[info] Iteration   3: 143,131 ops/s
[info] Iteration   4: 135,936 ops/s
[info] Iteration   5: 137,563 ops/s
[info] Iteration   6: 118,467 ops/s
[info] Iteration   7: 126,278 ops/s
[info] Iteration   8: 130,316 ops/s
[info] Iteration   9: 157,347 ops/s
[info] Iteration  10: 120,501 ops/s

Be aware, multi-threaded test can quickly put processor to throttling, resulting in large result times deviation. You may need to change parameters to avoid throttling, if you want to check how volatile on isClosed affects performance.

Reply:
Ok, looks like you are confident that this change isn't going to reduce perf. Let's proceed with it then